schema_validator.py

```
@@ -1,405 +1,346 @@
+        canonical = json.dumps(clone, ensure_ascii=False, sort_keys=True, separators=(",", ":"))
+        return hashlib.sha256(canonical.encode("utf-8")).hexdigest()
+
+    def _validate_with_schema(
+        self,
+        payload: Mapping[str, Any],
+        schema: Optional[Mapping[str, Any]],
+        report: SchemaValidationReport,
+        name: str,
+    ) -> None:
+        if not jsonschema or not schema:
+            report.add_warning(f"jsonschema missing – skipped schema validation for {name}")
```

Ruff (0.14.1)

String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)